### PR TITLE
[P2 tests] [Team 2] Add Overall: Large Data Stress Test

### PR DESCRIPTION
**Pull request by FuyaoLi2017**
What's your team number?
Team 2

What is the functionality being tested?
Overall Large Data Stress Test

Describe your tests briefly:
The test add the novel "pride-and-prejudice" as one document(708KB) for 5000 times and added two additional test documents. The whole test data is about 3.46GB.
"test1" tests searchQuery with keyword "elizabeth" and see if it has appeared in the documents for 5001 times. If it is correct, it means the engine can handle very large datasets.
"test2" tests searchAndQuery with keyword "qwertyuiop" and "elizabeth" and see if it finds 1 record. If it passes the test, it means the searchAndQuery() works well in large datasets.
"test3" tests searchOrQuery with keyword "qwertyuiop" and "elizabeth" and see if it finds 5002 record. If it passes the test, it means the searchOrQuery() works well in large datasets.

Does each test case have comments/documentation?
Yes

### CODE REVIEW

File: src/test/java/edu/uci/ics/cs221/index/Team2StressTest/Team2StressTest.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        List<String> keywords = Arrays.asList("qwertyuiop", "elizabeth");
+
+        Iterator<Document> result = invertedIndexManager.searchAndQuery(keywords);
+
+        int counter = 0;
+        while (result.hasNext()) {
+            counter++;
+            result.next();
+        }
+        assertEquals(1, counter);
```
Comment (cyy12345) [inline src/test/java/edu/uci/ics/cs221/index/Team2StressTest/Team2StressTest.java]: Will it have the word "qwertyuiop"?   I think the counter would be 0?

**Comment by cyy12345** [comment]
Very good test case. However for the file deletion, can we assume under the directory we only have some files (not directory)? Then we may can simplify the after process.

File: src/test/java/edu/uci/ics/cs221/index/Team2StressTest/Team2StressTest.java (hunk 1 above)
Comment (FuyaoLi2017) [reply src/test/java/edu/uci/ics/cs221/index/Team2StressTest/Team2StressTest.java]: It is added in line 54.

Comment (cyy12345) [reply src/test/java/edu/uci/ics/cs221/index/Team2StressTest/Team2StressTest.java]: Sorry for the mistake

**Comment by FuyaoLi2017** [comment]
> Very good test case. However for the file deletion, can we assume under the directory we only have some files (not directory)? Then we may can simplify the after process.

I have updated the code. Assuming the folder don't have any sub directories.

**Comment by ZTShao** [comment]
Very clear logic and good implementation of loading text from the online resources. The size of documents is over 3GB and I think this is quite enough for a stress test.

File: src/test/java/edu/uci/ics/cs221/index/Team2StressTest/Team2StressTest.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
+package edu.uci.ics.cs221.index.Team2StressTest;
```
Comment (LaytonW) [inline src/test/java/edu/uci/ics/cs221/index/Team2StressTest/Team2StressTest.java]: All test cases should be in the package `edu.uci.ics.cs221.index.inverted`.
See [TA's instruction on Slack](https://uci-cs221-s19.slack.com/archives/CH7JR156D/p1556138044128600).
```suggestion
package edu.uci.ics.cs221.index.inverted;
```
and move this file to `.../index/inverted/` accordingly.

Comment (FuyaoLi2017) [reply src/test/java/edu/uci/ics/cs221/index/Team2StressTest/Team2StressTest.java]: Thanks. I have updated the file location.

**Comment by FuyaoLi2017** [comment]
> ## Review from Team 1
> Well constructed test cases. Real-world large documents are used.
> 
> #### Minor issues:
> * Package needs to be corrected for consistency.
> * ❓ Time limit: I believe we do need to put time limits on test cases, especially these stress tests involving huge datasets. Otherwise, the execution time of these test cases may go unbounded. Maybe TA could give advises on the threshold @zuozhiw.

I don't know what time should be proper to handle this. Maybe TA could give me some suggestions. @zuozhiw

**Pull request merged**